documentation.md

```diff
@@ -0,0 +1,97 @@
+# Turbo for Django
+
+This repository aims to provide utilities for working with [Turbo](https://turbo.hotwire.dev)
+with the Django web framework. 
+
+## Turbo Streams
+Currently, the repository contains utilities for working with
+[Turbo Streams](https://turbo.hotwire.dev/handbook/streams) over Websockets, the one part of
+Turbo which requires a specific integration with the backend framework to function. In Django's
+case, this means taking advantage of [Django Channels](https://github.com/django/channels).
+
+The Django Streams integration is heavily inspired by the helpers and mixins found in
+[turbo-rails](https://github.com/hotwired/turbo-rails). There are a few steps required to get
+the integration working.
+
+### Initial Setup
+#### Django Channels
+After setting up Channels according to the [documentation](https://channels.readthedocs.io/en/stable/installation.html),
+make sure to modify the top-level `ProtocolTypeRouter` to include the `TurboStreamsConsumer`:
+
+```python
+from turbo.consumers import TurboStreamsConsumer
+application = ProtocolTypeRouter({
+  "http": AsgiHandler(),
+  "websocket": TurboStreamsConsumer.as_asgi()  # Leave off .as_asgi() if using Channels 2.x
+})
+```
+
+#### Including JavaScript
+In our `<head>` for all our templates, we need to `{% include 'turbo/head.html' %}`, which will include the Turbo libraries we need
+in order to use Turbo Drive, Frames and Streams in our app.
```

Comment:
Since this is required for even non-stream support, maybe it could be split out? 
Maybe a structure like (some of these could just be placeholders for now):

- Introduction
- Define terms (Frames, Streams, Drive) or just point to official docs
- Install
- - Base install (add `head` template)
- - Stream install (Optional?)
- Usage
- - Frames
- - Streams
- Production notes?

It might also be good to call out that Stream support requires ASGI?